third_party/move/move-compiler/src/expansion/dependency_ordering.rs

```diff
@@ -32,19 +32,32 @@ pub fn verify(
         addresses_by_node,
         ..
     } = context;
-    let graph = dependency_graph(&module_neighbors);
-    match petgraph_toposort(&graph, None) {
-        Err(cycle_node) => {
-            let cycle_ident = *cycle_node.node_id();
-            let error = cycle_error(&module_neighbors, cycle_ident);
-            compilation_env.add_diag(error);
-        },
-        Ok(ordered_ids) => {
-            for (order, mident) in ordered_ids.iter().rev().enumerate() {
-                modules.get_mut(mident).unwrap().dependency_order = order;
-            }
-        },
+    let imm_module_idents = modules
```

Comment:
I'm wondering, the DIGraph already gives as a topological sorting algorithm, right? Why then not just add a new edge vector -> module for each module in the graph which is _not_ in a dependency for vector (so we do not create a cycle). After that the topsort should just produce a sequence of sets of modules, one can use lexicographical ordering inside each set if this is important, but vector and all its dependencies would come before anything else.

Reply:
Yes, what you suggest is also a solution.

By adding edges from each (non vector module dependency) module to vector, we can end up adding hundreds of edges (in the case of aptos-framework, over a 100 edges). So, this can be a bit inefficient.

In my original solution, I extracted out the vector transitive closure out of the dependency graph, numbered it first, and then numbered the remaining dependency graph (numbering appropriately offset). This also guaranteed vector numbered lower than other modules.

While my original solution is more efficient, the logic may be a bit obscured. Thus, I have implemented the solution you proposed as it results in a smaller change and more straightforward to reason about.